components/clp-py-utils/clp_py_utils/s3_utils.py

```
@@ -392,3 +384,134 @@ def _gen_deletion_config(objects_list: List[str]):
+    :raises: ValueError if the object doesn't exist or fails to read the metadata.
+    :raises: Propagates `boto3.client.head_object`'s exceptions.
+    """
+    try:
+        return FileMetadata(
+            Path(key), s3_client.head_object(Bucket=bucket, Key=key)["ContentLength"]
+        )
+    except botocore.exceptions.ClientError as e:
+        raise ValueError(
+            f"Failed to read metadata of the key `{key}` from the bucket `{bucket}`"
+            f" with the error: {e}."
+        )
```

Ruff (0.14.0)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)